lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1985,6 +1985,14 @@ ArrayAttr ClassOp::getLayersAttr() { return ArrayAttr::get(getContext(), {}); }
 
 ArrayRef<Attribute> ClassOp::getLayers() { return getLayersAttr(); }
 
+void ClassOp::setLayersAttr(ArrayAttr layers) {
+  assert(layers.empty() && "classes may not enable layers");
+}
+
+void ClassOp::setLayers(ArrayRef<Attribute> layers) {
+  assert(layers.empty() && "classes may not enable layers");
+}
```

Comment:
This isn't great (see comments above). This will silently do nothing with asserts off. Maybe that's okay?

Reply:
Modulo above discussion (unsure), I think it's probably okay to only report you used the API badly in asserts-enabled builds.

At least one MLIR interface has the default implementation for the "set" version of a thing simply do nothing, so there's that option too.  I like the assert "JIC" though :+1: .